crypto/rsa/rsa_lib.c

```diff
@@ -972,20 +980,20 @@ int EVP_PKEY_CTX_set_rsa_mgf1_md_name(EVP_PKEY_CTX *ctx, const char *mdname,
             && ctx->pmeth->pkey_id != EVP_PKEY_RSA_PSS)
         return -1;
 
-    *p++ = OSSL_PARAM_construct_utf8_string(OSSL_ASYM_CIPHER_PARAM_MGF1_DIGEST,
+    *p++ = OSSL_PARAM_construct_utf8_string(OSSL_PKEY_PARAM_MGF1_DIGEST,
```

Comment:
Why does the legacy check on line 970 look different to the previous checks (that also check for NULL)

Reply:
I...  don't know, and not sure I see clearly what you're talking about...